### PR TITLE
Upgrade resolver and some other clean ups

### PR DESCRIPTION
**Pull request by L0neGamer**
Happy to drop commits from this, but I wanted to take the opportunity while I was upping the resolver and moving on from pre aeson-2 to do a little bit of clean up for some things I noticed.

### CODE REVIEW

File: src/Discord/Internal/Types/Channel.hs [hunk 1]
```diff
@@ -170,7 +170,7 @@ instance FromJSON Channel where
     case type' of
       0 ->
         ChannelText  <$> o .:  "id"
-                     <*> o .:? "guild_id" .!= 0
```
Comment (aquarial) [inline src/Discord/Internal/Types/Channel.hs:173]: Good idea to remove the default handling. A missing guild_id should be a crash

File: src/Discord/Internal/Types/Prelude.hs [hunk 2]
```diff
@@ -301,15 +285,15 @@ objectFromMaybes = object . catMaybes
 --
 -- Public creation of this datatype should be done using the relevant smart
 -- constructors for Emoji, Sticker, or Avatar.
-data Base64Image a = Base64Image T.Text T.Text
+data Base64Image a = Base64Image { mimeType :: T.Text, base64Data :: B.StrictByteString }
```
Comment (aquarial) [inline src/Discord/Internal/Types/Prelude.hs:288]: B.StrictByteString doesn't exist in cabal 8.10 (generating a github actions build error). It's just a type alias for B.ByteString (TIL). 

Should probably use B.ByteString for compatibility?

Comment (L0neGamer) [reply src/Discord/Internal/Types/Prelude.hs]: Made this adjustment. Odd that the names are like that, yeah. Personally I'd make StrictByteString the actual name so it's easier to tell it and LazyByteString apart, but I'm not a maintainer there...

**Review by aquarial — APPROVED** [review]
Looks good! 



Let's drop Aeson <2

**Pull request merged**